hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -1641,7 +1641,7 @@ void optimizeARM(Vunit& unit, const Abi& abi, bool regalloc) {
   simplify(unit);
 
   lowerForARM(unit);
-
+  reuseImmq(unit);
```

Comment:
Looks like there's something in the vixl repo that can be embraced.

    369 int MacroAssembler::MoveImmediateHelper(MacroAssembler* masm,
    370                                         const Register& rd,
    371                                         uint64_t imm) {
    372   bool emit_code = (masm != NULL);
src/vixl/a64/macro-assembler-a64.cc